### ISSUE TITLE
It shouldn't be possible to capture reassignable variables within an inflight scope

### ISSUE THREAD

**Issue by eladb** [winglang/wing#2729]
### I tried this:

```js
bring cloud;

test "inner inflight class capture immutable" {
  let y = MutArray<num>[1];
  let var i = 10;

  class Inner {
    dang(): num {
      i = i + 1;
      y.push(2);
      return y.at(0) + 10;
    }
  }

  assert(new Inner().dang() == 11);
  assert(y.at(1) == 2);
  assert(i == 11);
}
```

### This happened:

The last assertion failed (`i` was still 10 at this point).

### I expected this:

We should forbid capturing reassignable variables. We can be opinionated about this because I believe this usually represent bad design.

I am okay with allowing capturing mutable objects within an inflight scope (because we are running within the same memory space), so the first case works.

### Component

Compiler

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by staycoolcall911** [comment]
Duplicate of #1908 (which is p3 btw).

**Comment by staycoolcall911** [comment]
Closed in #3213
